dpctl/tensor/_manipulation_functions.py

```diff
@@ -38,6 +38,43 @@ def _broadcast_strides(X_shape, X_strides, res_ndim):
     return tuple(out_strides)
 
 
+def _broadcast_shapes(*args):
+    """
+    Broadcast the input shapes into a single shape;
+    returns tuple broadcasted shape.
+    """
+    shapes = [array.shape for array in args]
+    if len(set(shapes)) == 1:
+        return shapes[0]
+    shapes = [list(shape) for shape in shapes]
```

Comment:
This upfront conversion should be delayed until necessary.

Reply:
I removed this line and changed to make the copy into mutable structure only when/if needed